includes/data-events/connectors/ga4/class-ga4.php

```diff
@@ -139,6 +147,44 @@ public static function filter_event_body( $body, $event_name ) {
 
 	}
 
+	/**
+	 * When the donation_new event is dispatched from the Stripe webhook, it can't catch GA client ID from Cookies.
```

Comment:
This is also an issue with other logic inside `GA4::filter_event_body()`. The event triggered by Stripe webhook is being dispatched with `logged_in` set to `no` and without `is_reader`, even though I'm a logged-in reader.

Reply:
I was tired when I noticed this the other day.

Looking at this again today I was happy to see it was an easy fix after all I did.

I also figured we could default the value of `is_reader` to `no` if the user is not logged in

https://github.com/Automattic/newspack-plugin/pull/2337/commits/df620d276bb0a0417399b94fb4b997f1e6c3f576